Automattic/newspack-plugin
# Update WordPress dependencies

### All Submissions:






### Changes proposed in this Pull Request:


Over at wp-calypso, we're updating to React 17. This means getting our peer dependencies updated to the right versions, and it turns out that `newspack-components` is a peer dep of `newspack-blocks`, which we sync to WordPress.com via Calypso.

This PR updates to React 17 and to the latest WordPress component packages. 

Also updates most NPM packages to the latest compatible versions.

Also specifies Node version support as `v16.11.1`, matching Calypso's Node support.

Also enables Dependabot to keep NPM and Composer packages up-to-date.

**Bonus:** Also fixes a minor pet peeve of mine, which is that this repo's `npm run release:archive` script previously built the plugin archive ZIP to the `assets/release` directory, whereas every other Newspack plugin repo builds to `./release` (in the root). Now this repo should also build to the root `./release` directory.

### How to test the changes in this Pull Request:

1. Using [NVM](https://github.com/nvm-sh/nvm), set your current Node version to v16.11.1 (matching Calypso's supported version)
2. Ensure `npm ci` runs
3. Ensure all NPM build, lint and test scripts still run (`build`, `dev`, `lint`, `test`)
4. Test newspack components (see Components Demo page)
5. Test newspack plugin (especially wizard pages)

### Possible issues/follow-up

- [x] ~Dependency `react-daterange-picker` is stale and doesn't appear to be updated anymore. It doesn't list react 17 support, but I suppose it probably supports it anyways.~ The package still works in the Campaigns Analytics page, so I don't think we need to update this immediately.
- [x] ~Build doesn't work (possible incompatibility with the node version I'm using)~ Should now work with Node v16.11.1

### Other information:
